### ISSUE TITLE
Support for the new CoordinatorLayout

### ISSUE THREAD

**Issue by lynfogeek** [futuresimple/android-floating-action-button#209]
WIth the new Design library from Google comes the CoordinatorLayout that allows the official FAB to react to different events such as scrolling or the entrance/dismiss  of a Snackbar below the FAB.

Such behavior would be great, as this library still handle many more features than the official FAB.

**Comment by CodeBendr** [comment]
Is this currently available?!

**Comment by lynfogeek** [comment]
@CodeBendr of course! http://android-developers.blogspot.nl/2015/05/android-design-support-library.html

**Comment by chalup** [comment]
I'll definitely do this.

**Comment by mostafazh** [comment]
+1

**Comment by scarletsky** [comment]
+1
This is really useful!

**Comment by nicktogo** [comment]
I use Timer as a workaround, it seems ridiculous. I will put my code here, hoping it can help someone:)
```
//get your FloatingActionMenu or FloatingActionButton
FloatingActionMenu menu = (FloatingActionsMenu) findViewById(R.id.menu);
//show your Snackbar
Snackbar.make(parentLayout, "Selected!", Snackbar.LENGTH_SHORT).show();
//move the menu up
menu.animate().translationYBy(-80);
//set a Timer to move the menu down
Timer timer = new Timer();
timer.schedule(new TimerTask() {
       @Override
       public void run() {
             menu.animate().translationYBy(80);
       }
 }, 1800);//Snackbar's short duration is 1500ms, while the long one is 2750ms, adjust the delay of timer to fit the moment the Snackbar dismiss
```

**Comment by xdbas** [comment]
When is this feature expected to be implemented?

**Comment by Ph1b** [comment]
Ill implemented it myself in the past. Going to make a pull request if that is wanted. But that will involve the support library as a dependency. Is that wanted? Else this can't be implemented.

**Comment by chalup** [comment]
@Ph1b Ideally this should be an optional dependency, but AFAIK gradle doesn't support it. As a workaround we can create a submodule with Behavior implementation. The only downside is that we won't be able to specify DefaultBehavior for FAB/FAM.

**Comment by Ph1b** [comment]
I could also just provide the java file and you can include it how ever you like. Just tell me.

**Comment by chalup** [comment]
@Ph1b Open PR or put it in gist, I'll think about the best way to include it.

**Comment by Ph1b** [comment]
It seems I have less time than I thought ;-)

So [here we go](https://github.com/Ph1b/MaterialAudiobookPlayer/blob/develop/audiobook/src/main/java/de/ph1b/audiobook/uitools/FabBehavior.java). I aggree to license my changes under your license. But as it was mostly a copy and paste I don't think I have too much a creative licensing right ;-)
It's basically the ported behavior of googles FAB for the SnackBar only. Building upon that one could also implement the other layout types easily.

**Comment by 0jickson0** [comment]
link is broken :(

**Comment by Ph1b** [comment]
[history](https://github.com/Ph1b/MaterialAudiobookPlayer/blob/v2.2.3.1/audiobook/src/main/java/de/ph1b/audiobook/uitools/FabBehavior.java)

**Comment by vergauwen-simon** [comment]
@Ph1b how must I set the behaviour to the FAB?

**Comment by MangeshGhotage** [comment]
Waiting for this feature

**Comment by Ph1b** [comment]
@vergauwen-simon 
In xml by 
`app:layout_behavior="de.ph1b.audiobook.uitools.FabBehavior"`

**Comment by AndroidDeveloperLB** [comment]
@nicktogo You can use Handler instead.

**Comment by sandro-csimas** [comment]
+1

**Comment by sakuyayc** [comment]
here is a solution: https://lab.getbase.com/introduction-to-coordinator-layout-on-android/

**Comment by PaulWoitaschek** [comment]
@sakuyayc 

I posted the solution already on 25th of July 2015 but it was not yet included into the library.

**Comment by nakirinakuru** [comment]
Thanks for the PR, it works

**Comment by ja-kub** [comment]
@chalup will you include PaulWoitaschek's PR?
